contribs/gnodev/setup_node.go

```diff
@@ -12,9 +12,45 @@ import (
 	"github.com/gnolang/gno/contribs/gnodev/pkg/emitter"
 	"github.com/gnolang/gno/contribs/gnodev/pkg/packages"
 	"github.com/gnolang/gno/gno.land/pkg/gnoland"
+	"github.com/gnolang/gno/gno.land/pkg/gnoland/ugnot"
+	"github.com/gnolang/gno/gno.land/pkg/sdk/vm"
 	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/std"
 )
 
+// extractDependenciesFromTxs extracts dependencies from transactions and adds them to the paths slice and config.BalancesList.
+func extractDependenciesFromTxs(nodeConfig *gnodev.NodeConfig, paths *[]string) {
+	for _, tx := range nodeConfig.InitialTxs {
+		for _, msg := range tx.Tx.Msgs {
+			// TODO: Support MsgRun
+			if callMsg, ok := msg.(vm.MsgCall); ok {
```

Comment:
nit: generally speaking, prefer early continue / return than nested indented code
```suggestion
             callMsg, ok := msg.(vm.MsgCall)
			if !ok {
			    continue
			}
```